lib/code_corps/github/sync/comment/comment/comment.ex

```diff
@@ -23,62 +24,58 @@ defmodule CodeCorps.GitHub.Sync.Comment.Comment do
     Task,
     User
   }
-  alias CodeCorps.GitHub.Sync.Comment.Comment.Changeset, as: CommentChangeset
   alias Ecto.Changeset
 
   @type outcome :: {:ok, list(Comment.t)} |
                    {:error, {list(Comment.t), list(Changeset.t)}}
 
   @doc ~S"""
-  Creates or updates `CodeCorps.Comment` records for the specified list of
-  `CodeCorps.Task` records.
-
-  When provided a list of `CodeCorps.Task` records, a `CodeCorps.GithubComment`,
-  a `CodeCorps.User`, and a GitHub API payload , for each `CodeCorps.Task`
-  record, it creates or updates a `CodeCorps.Comment` record, using the provided
-  GitHub API payload, associated to the specified `CodeCorps.GithubComment` and
-  `CodeCorps.User`
+  Creates or updates a `CodeCorps.Comment` for the specified `CodeCorps.Task`.
+
+  When provided a `CodeCorps.Task`, a `CodeCorps.GithubComment`, a
+  `CodeCorps.User`, and a GitHub API payload, it creates or updates a
+  `CodeCorps.Comment` record, using the provided GitHub API
+  payload, associated to the specified `CodeCorps.GithubComment`,
+  `CodeCorps.Task` and `CodeCorps.User`
   """
-  @spec sync_all(list(Task.t), GithubComment.t, User.t, map) :: outcome
-  def sync_all(tasks, %GithubComment{} = github_comment, %User{} = user, %{} = payload) do
-    tasks
-    |> Enum.map(&sync(&1, github_comment, user, payload))
-    |> ResultAggregator.aggregate
+  @spec sync_task(Task.t, GithubComment.t, User.t) :: outcome
+  def sync_task(%Task{} = task, %GithubComment{} = github_comment, %User{} = user) do
+    task
+    |> find_or_init_comment(github_comment)
+    |> Sync.Comment.Comment.Changeset.build_changeset(github_comment, task, user)
+    |> Repo.insert_or_update()
   end
 
   @doc ~S"""
-  Creates or updates `Comment` records for the specified `ProjectGithubRepo`.
+  Creates or updates `CodeCorps.Comment` records for the specified
+  `CodeCorps.ProjectGithubRepo`.
 
-  For each `GithubComment` record that relates to the `GithubRepo` for a given
-  `ProjectGithubRepo`:
+  For each `CodeCorps.GithubComment` record that relates to the
+  `CodeCorps.GithubRepo` for a given`CodeCorps.ProjectGithubRepo`:
 
-  - Find the related `Task` record
-  - Create or update the `Comment` record
-  - Associate the `Comment` record with the `User` that relates to the
-    `GithubUser` for the `GithubComment`
+  - Find the related `CodeCorps.Task` record
+  - Create or update the related `CodeCorps.Comment` record
+  - Associate the `CodeCorps.Comment` record with the `CodeCorps.User` that
+    relates to the `CodeCorps.GithubUser` for the `CodeCorps.GithubComment`
```

Comment:
Did you check how the docs generate? As I said in slack, they do not generate correctly for me when aliased. If the issue is just for me, then I'm ok with using aliased. Otherwise, looks like non-aliased is necessary.